src/components/Canvas/DataLayer.tsx

```diff
@@ -548,7 +555,106 @@ export default class DataLayer extends BaseLayer {
     return { swipedPixels, validColumnIndices, validRowIndices };
   }
 
+  updateCapturedImageBitmap() {
+    let squareLength = this.gridSquareLength * this.panZoom.scale;
+    // leftTopPoint is a cartesian coordinate
+    const allRowKeys = getRowKeysFromData(this.getData());
+    const allColumnKeys = getColumnKeysFromData(this.getData());
+    const imageWidth = allColumnKeys.length * squareLength;
+    const imageHeight = allRowKeys.length * squareLength;
+    const capturedPanZoomValue = {
+      offset: { ...this.panZoom.offset },
+      scale: this.panZoom.scale,
+    };
+    if (imageWidth > MaxImageBitMapSideLength) {
+      squareLength = MaxImageBitMapSideLength / allColumnKeys.length;
+      capturedPanZoomValue.scale = squareLength / this.gridSquareLength;
+    }
+    if (imageHeight > MaxImageBitMapSideLength) {
+      squareLength = MaxImageBitMapSideLength / allRowKeys.length;
+      capturedPanZoomValue.scale = squareLength / this.gridSquareLength;
+    }
+
+    const width = this.dpr
+      ? allColumnKeys.length * squareLength * this.dpr
+      : allColumnKeys.length * squareLength;
+    const height = this.dpr
+      ? allRowKeys.length * squareLength * this.dpr
+      : allRowKeys.length * squareLength;
+
+    const canvas = new OffscreenCanvas(width, height);
```

Comment:
The API seems to not cleanup some code since the API you provided is creating a canvas dom element and then changing it into a offscreencanvas. From what I know, `new OffscreenCavnas(width, height)` is the same thing but it is different in that it skips the creation of a canvas dom element.

Reply:
> is creating a canvas dom element and then changing it into a offscreencanvas

without creating a new dom element isn't it approachable by?

```js
const offscreenCanvas = this.element.transferControlToOffscreen();
```

Reply:
Let me try to implement it. I am not sure if it will interfere with the original contents in the element.

Reply:
I found that once you transfer control to offscreen, you cannot use it in the main thread. Thus I believe there is a need to create a new dom element if we want the worker thread to access the canvas and render it.